google/resource_bigtable_instance.go

```diff
@@ -207,6 +209,51 @@ func resourceBigtableInstanceRead(d *schema.ResourceData, meta interface{}) erro
 	return nil
 }
 
+func resourceBigtableInstanceUpdate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	ctx := context.Background()
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	c, err := config.bigtableClientFactory.NewInstanceAdminClient(project)
+	if err != nil {
+		return fmt.Errorf("Error starting instance admin client. %s", err)
+	}
+	defer c.Close()
+
+	if d.Get("instance_type").(string) == "DEVELOPMENT" {
+		return resourceBigtableInstanceRead(d, meta)
```

Comment:
Since `num_nodes` can't be updated for development clusters (or set), we should probably error instead of silently doing nothing. As-is, setting a value for `num_nodes` on an existing development cluster results in a permadiff.

Erroring here would cause an apply-time error. Alternatively, if we add a `CustomizeDiff` step we can error earlier, at plan time. We can  check for these failure conditions:

* the new value of `instance_type` is `DEVELOPMENT` and `new_count` != 1

* the new value of `instance_type` is `DEVELOPMENT` and `cluster.0.num_nodes` != `nil` or `0` (I believe it'll always be 0, but I'm not 100% certain of that.)

Reply:
Sounds good to me, will update. Seems like we can rely on `0` being the default based on this https://github.com/hashicorp/terraform/blob/39b6a322e67625ed641695821b25dd718f50ca9a/helper/schema/schema.go#L1837-L1838